src/jit/codegenxarch.cpp

```diff
@@ -3921,41 +3999,22 @@ void CodeGen::genStructPutArgUnroll(GenTreePutArgStk* putArgNode, unsigned baseV
     // Fill the remainder (15 bytes or less) if there's one.
     if ((size & 0xf) != 0)
     {
-        // Grab the integer temp register to emit the remaining loads and stores.
-        regNumber tmpReg = genRegNumFromMask(putArgNode->gtRsvdRegs & RBM_ALLINT);
-        assert(genIsValidIntReg(tmpReg));
-
-        if ((size & 8) != 0)
-        {
-            genCodeForLoadOffset(INS_mov, EA_8BYTE, tmpReg, src->gtOp.gtOp1, offset);
-
-            emit->emitIns_S_R(INS_mov, EA_8BYTE, tmpReg, baseVarNum, putArgOffset + offset);
-
-            offset += 8;
-        }
-
-        if ((size & 4) != 0)
-        {
-            genCodeForLoadOffset(INS_mov, EA_4BYTE, tmpReg, src->gtOp.gtOp1, offset);
-
-            emit->emitIns_S_R(INS_mov, EA_4BYTE, tmpReg, baseVarNum, putArgOffset + offset);
-
-            offset += 4;
-        }
-
-        if ((size & 2) != 0)
+#ifdef _TARGET_X86_
+        if (m_pushStkArg)
         {
-            genCodeForLoadOffset(INS_mov, EA_2BYTE, tmpReg, src->gtOp.gtOp1, offset);
-
-            emit->emitIns_S_R(INS_mov, EA_2BYTE, tmpReg, baseVarNum, putArgOffset + offset);
-
-            offset += 2;
+            offset = (size & 0xf);
+            offset -= genMove8IfNeeded(size, longTmpReg, src->gtOp.gtOp1, offset - 8);
+            offset -= genMove4IfNeeded(size, intTmpReg, src->gtOp.gtOp1, offset - 4);
+            offset -= genMove2IfNeeded(size, intTmpReg, src->gtOp.gtOp1, offset - 2);
+            offset -= genMove1IfNeeded(size, intTmpReg, src->gtOp.gtOp1, offset - 1);
```

Comment:
It seems like the moves will use odd numbers  for "offset -8", "offset -4" etc..
Are these going to be unaligned stores. into the stack?
We should  try to avoid using unaligned stores if we can